gradle-diff-task-dependencies/plugin/src/main/groovy/com/novoda/gradle/diffdependencytask/git/ChangedFilesProvider.groovy

```diff
@@ -0,0 +1,42 @@
+package com.novoda.gradle.diffdependencytask.git
+
+class ChangedFilesProvider implements GroovyCallable<List<String>> {
+
+    private final GroovyCallable<String> getCurrentBranch
+    private final GroovyCallable<String> getTargetBranch
+
+    private List<String> changedFiles
+
+    static ChangedFilesProvider create() {
+        return new ChangedFilesProvider(new CurrentBranchProvider(), new TargetBranchProvider())
+    }
+
+    ChangedFilesProvider(GroovyCallable<String> currentBranchProvider,
+                         GroovyCallable<String> targetBranchProvider) {
+
+        this.getCurrentBranch = currentBranchProvider
+        this.getTargetBranch = targetBranchProvider
+    }
+
+    private synchronized List<String> getChangedFiles() {
```

Comment:
I was wondering why the `synchronized` here (and below). Are you afraid this will be executed in concurrent threads?